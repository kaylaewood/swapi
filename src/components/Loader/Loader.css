.loader-image {
  animation: spinning-pic infinite 10s linear;
  width: 30vw;
}

@keyframes spinning-pic {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.loader-text {
  color: white;
  font-size: 3vw;
  margin-top: .5vw;
}

.loader-container {
  margin: auto;
  margin-top: 5vw;
  text-align: center;
}
