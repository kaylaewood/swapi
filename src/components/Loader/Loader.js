import React from 'react';
import deathStar from '../../images/death-star.png';
import './Loader.css'

const Loader = () => (
<div className='loader-container'>
  <img src={deathStar} className="loader-image" alt="deathStar" />
  <p className='loader-text'>loading...</p>
</div>
)

export default Loader;
