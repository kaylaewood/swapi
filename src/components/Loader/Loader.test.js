import React from 'react';
import ReactDOM from 'react-dom';
import Loader from './Loader';
import { shallow } from 'enzyme';

describe('Loader', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Loader />)
    expect(wrapper).toMatchSnapshot();
    })
})
