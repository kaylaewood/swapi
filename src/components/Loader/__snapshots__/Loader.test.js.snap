// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should match the snapshot 1`] = `
<div
  className="loader-container"
>
  <img
    alt="deathStar"
    className="loader-image"
    src="death-star.png"
  />
  <p
    className="loader-text"
  >
    loading...
  </p>
</div>
`;
